tests/containers/podman_integration.pm

```diff
@@ -0,0 +1,70 @@
+# SUSE's openQA tests
+#
+# Copyright 2024 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Package: podman integration
+# Summary: Upstream podman integration tests
+# Maintainer: QE-C team <qa-c@suse.de>
+
+use Mojo::Base 'containers::basetest';
+use testapi;
+use serial_terminal qw(select_serial_terminal select_user_serial_terminal);
+use containers::utils qw(get_podman_version);
+use utils qw(zypper_call);
+use version_utils qw(get_os_release);
+use containers::common;
+
+sub run {
+    select_serial_terminal;
+    my ($running_version, $sp, $host_distri) = get_os_release;
+    install_podman_when_needed($host_distri);
+
+    # Install tests dependencies
+    my @pkgs = qw(apache2-utils bats buildah criu go gpg2 jq make netcat-openbsd openssl podman-remote python3-PyYAML skopeo socat sudo systemd-container);
+    zypper_call "in @pkgs";
+
+    my $podman_bats = get_required_var("PODMAN_BATS");
+    my $opts = "";
+    if ($podman_bats =~ /^rootless/) {
+        select_user_serial_terminal();
+        $opts = "--rootless";
+    } else {
+        $opts = "--root";
+    }
+    $opts .= " --remote" if ($podman_bats =~ /remote/);
+
+    my $podman_version = get_podman_version();
+    assert_script_run "curl -sL https://github.com/containers/podman/archive/refs/tags/v$podman_version.tar.gz | tar -zxf -";
+    assert_script_run "cd podman-$podman_version/";
+
+    # NOTE: These tests are known to fail.  Needs investigation.
+    my @skip_tests = qw(120-load.bats 125-import.bats 250-systemd.bats 255-auto-update.bats 520-checkpoint.bats);
+    foreach my $test (@skip_tests) {
+        assert_script_run "rm -f test/system/$test";
+    }
+    if ($podman_bats =~ /remote/) {
+        background_script_run "podman system service --timeout=0";
+    }
+
+    assert_script_run "env PODMAN=/usr/bin/podman QUADLET=/usr/libexec/podman/quadlet hack/bats $opts |& tee /tmp/bats.txt", 2600;
```

Comment:
I recommend not using hardcoded paths such as `/tmp/bats.txt` in `/tmp`, as this can be a security issue (`/tmp` is world-writable) and it can also conflict with other modules or even the same module run again in a different mode later.

Reply:
Using the CWD here would probably work fine, i.e. just `bats.txt`